### DISCUSSION TITLE
A consolidated report from multiple agents

### DISCUSSION THREAD

**Discussion by eastclintw00d** [rstudio/pointblank#337] [Q&A] [answered]
In my ETL routines I often build multiple tables in a single script. I would like to run agents on all of them and afterwards get a consolidated report. Is that possible? I did not find a clear statement in that direction.

**Comment by rich-iannone** [comment, marked as answer]
This is now possible in the development version of the package with the improved `get_multiagent_report()` function. The docs at https://rich-iannone.github.io/pointblank/reference/get_multiagent_report.html explain how the function works now. Basically it's create several agents (and `interrogate()`), create a `multiagent` with `create_multiagent()`, then either print the object or use `get_multiagent_report()` to customize the report output.

Let me know how this works out for you!

**Reply by eastclintw00d** [reply]
Thanks @rich-iannone,

It does almost what I want. The problem in my case is that the tables are completely unrelated. According to the link you posted there should be an option to set  `display_mode = "long"` while, in fact, in version 0.7.0 the option seems to be removed and `display_mode = "wide"` is set.

![grafik](https://user-images.githubusercontent.com/7422098/125597999-f36ce8fc-5d22-4162-9520-24069aba6bea.png)

It would also be great if there was a method for `all_passed()` applied to a multiagent object.

**Reply by rich-iannone** [reply]
@eastclintw00d What you need currently is the dev version of the package (this feature is not in 0.7.0). So use `devtools::install_github("rich-iannone/pointblank")` to get that. Pretty soon (maybe 2-3 weeks from now), pointblank 0.8.0 will be released so the long report style will be available in the CRAN version.

**Reply by rich-iannone** [reply]
@eastclintw00d I like your idea about making `all_passed()` work with the multiagent. I'll file an issue to get that work prioritized.

**Reply by eastclintw00d** [reply]
@rich-iannone 

Thanks, your github version provides exactly what I was looking for.
